loki/transformations/tests/test_loop_blocking.py

```diff
@@ -0,0 +1,538 @@
+def test_normalized_loop_range(tmp_path, frontend):
+    """
+    Tests the num_iterations and normalized_loop_range functions.
+    """
+    for start in range(-10, 11):
+        for stop in range(start + 1, 50 + 1, 4):
+            for step in range(1, stop - start):
+                loop_range = LoopRange((start, stop, step))
+                pyrange = range(start, stop + 1, step)
+
+                normalized_range = normalized_loop_range(loop_range)
+                assert normalized_range.step is None, "LoopRange.step should be None in a normalized range"
+
+                normalized_start = LokiEvaluationMapper()(normalized_range.start)
+                assert normalized_start == 1, "LoopRange.start should be equal to 1 in a normalized range"
+
+                normalized_stop = floor(LokiEvaluationMapper()(normalized_range.stop))
+                assert normalized_stop == len(
+                    pyrange), "LoopRange.stop should be equal to the total number of iterations of the original LoopRange"
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_iteration_number(tmp_path, frontend):
+    for start in range(-10, 11):
+        for stop in range(start + 1, 50, 4):
+            for step in range(1, stop - start):
+                loop_range = LoopRange((start, stop, step))
+                pyrange = range(start, stop + 1, step)
+                normalized_range = get_pyrange(normalized_loop_range(loop_range))
+                assert len(normalized_range) == len(
+                    pyrange), "Length of normalized loop range should equal length of python loop range"
+
+                LEM = LokiEvaluationMapper()
+                assert all(n == LEM(iteration_number(IntLiteral(i), loop_range)) for i, n in
+                           zip(pyrange, normalized_range))
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_iteration_index(tmp_path, frontend):
+    for start in range(-10, 11):
+        for stop in range(start + 1, 50, 4):
+            for step in range(1, stop - start):
+                loop_range = LoopRange((start, stop, step))
+                pyrange = range(start, stop + 1, step)
+                normalized_range = get_pyrange(normalized_loop_range(loop_range))
+                assert len(normalized_range) == len(
+                    pyrange), "Length of normalized loop range should equal length of python loop range"
+
+                LEM = LokiEvaluationMapper()
+                assert all(i == LEM(iteration_index(IntLiteral(n), loop_range)) for i, n in
+                           zip(pyrange, normalized_range))
+
+
+
+
+
+
+"""
+Splitting tests. These tests check that loop
+"""
+LOKI_LOOP_SLIT_VAR_ADDITION = 7
+
+@pytest.mark.parametrize('frontend', available_frontends())
+# @pytest.mark.parametrize('block_size', [10, 117, 250])
+# @pytest.mark.parametrize('n', [50, 193, 500, 1200])
+@pytest.mark.parametrize('block_size', [10])
+@pytest.mark.parametrize('n', [50])
```

Comment:
Debug leftover?

```suggestion
@pytest.mark.parametrize('block_size', [10, 117, 250])
@pytest.mark.parametrize('n', [50, 193, 500, 1200])
```

Reply:
Yes, thank you for spotting it. I am not sure how many block_sizes and iterations to test for. It is recompiling for each case in the matrix so the test time increases substantially if we test too many options.